DEVELOPER.md

```diff
@@ -1,4 +1,110 @@
-##TODO - more info
+# Building and Testing ng-bootstrap
+
+This document describes how to set up your development environment to build and test ng-bootstrap.
+It also explains the basic mechanics of using `git`, `node`, and `npm`.
+
+* [Prerequisite Software](#prerequisite-software)
+* [Getting the Sources](#getting-the-sources)
+* [Installing NPM Modules](#installing-npm-modules)
+* [Build commands](#build-commands)
+* [Running Tests Locally](#running-tests-locally)
+* [Formatting](#clang-format)
+
+See the [contribution guidelines](https://github.com/ng-bootstrap/core/blob/master/CONTRIBUTING.md)
+if you'd like to contribute to ng-bootstrap.
+
+## Prerequisite Software
+
+Before you can build and test ng-bootstrap, you must install and configure the
+following products on your development machine:
+
+* [Git](http://git-scm.com) and/or the **GitHub app** (for [Mac](http://mac.github.com) or
+  [Windows](http://windows.github.com)); [GitHub's Guide to Installing
+  Git](https://help.github.com/articles/set-up-git) is a good source of information.
+
+* [Node.js](http://nodejs.org), (version `>=4.2.1 <5`) which is used to run tests, and generate distributable files. We also use Node's Package Manager, `npm` 
+  (version `>=2.14.7 <3.0`), which comes with Node. Depending on your system, you can install Node either from 
+  source or as a pre-packaged bundle.
+
+* [Chrome](https://www.google.es/chrome/browser/desktop/index.html), we use Chrome to run our tests.
+
+## Getting the Sources
+
+Fork and clone the ng-bootstrap repository:
+
+1. Login to your GitHub account or create one by following the instructions given
+   [here](https://github.com/signup/free).
+2. [Fork](http://help.github.com/forking) the [main ng-bootstrap
+   repository](https://github.com/ng-bootstrap/core).
+3. Clone your fork of the ng-bootstrap's core repository and define an `upstream` remote pointing back to
+   the ng-bootstrap's core repository that you forked in the first place.
+
+```shell
+# Clone your GitHub repository:
+git clone git@github.com:<github username>/core.git
```

Comment:
This should maybe be `git clone git@github.com:<github username>/core.git ng-bootstrap` so people have it in a better named directory.